### ISSUE TITLE
Enhancement: options to force gm to gs reset and capital tone fallback.

### ISSUE THREAD

**Issue by zaphod77** [nukeykt/Nuked-SC55#36]
While these are not very accurate to the real hardware, they will be useful for playing back broken midis.

basically i want the sc-55 mkII in forced gs mode with capital tone fallback, because it has more instruments.

Most non sound canvas synths that people may have had when working on midi files on their computers that have gs instrument variations always make them available, even after a GM reset, and they are of course broken on a real sound canvas.

Of course it should not be on by default, but this would probably be much appreciated by people trying to play back midi files that don't have a sequencer handy to fix them up. :)

**Comment by Grieferus** [comment]
As far as I know, current reset prompt (Q, hold U and press Q, W) is GS reset. As of Capital Tone Fallback, can't say anything about it.

**Comment by zaphod77** [comment]
i'm saying that a number of midi files have gm resets, but then use gs variation instruments.  which works only on the original sc-55. except that some of them use the extended mkii instruments. and this works on most "gs" soundfonts and wavetable cards that used them, but not on any real sound canvas at all.

i'm looking for a "just make the darn midi work right" button. :)

**Comment by nikitalita** [comment]
I remembered @Kitrinx talking about adding capital tone feedback: 
```
Im pretty sure I could turn any model of sc55 into one that can do capital tone fallback
the way their maps work, all you have to do is just fill all unused instruments with the capital tones
I wrote a program that modifies the instrument table in the control rom
then you can just flash it back to an eeprom
for capital tones, all you do is just copy the data from the regular instrument's jump table to any unused sub-instrument
it has the same effect as falling back
```

**Comment by zaphod77** [comment]
i'm definitely interested in a modified mkII rom that treats gm as gs reset and has capital tone fallback.  i'd be even more interested in it if i actually had hardware.

**Comment by Karmeck** [comment]
> I remembered @Kitrinx talking about adding capital tone feedback:
> 
> ```
> Im pretty sure I could turn any model of sc55 into one that can do capital tone fallback
> the way their maps work, all you have to do is just fill all unused instruments with the capital tones
> I wrote a program that modifies the instrument table in the control rom
> then you can just flash it back to an eeprom
> for capital tones, all you do is just copy the data from the regular instrument's jump table to any unused sub-instrument
> it has the same effect as falling back
> ```

Indeed he did

![Screenshot_20240414_151049_Discord](https://github.com/nukeykt/Nuked-SC55/assets/33399802/25b2c709-f5a9-4f9a-819a-641ede523364)
![Screenshot_20240414_151203_Discord](https://github.com/nukeykt/Nuked-SC55/assets/33399802/4cbf0833-196d-46b1-89bb-8c985bcb7d6e)


Here is the github page
https://github.com/Kitrinx/SC55_Soundfont

**Comment by zaphod77** [comment]
the github page doesn't appear to have the program. just the converter.

**Comment by Karmeck** [comment]
> the github page doesn't appear to have the program. just the converter.

It does not. As seen in the comments from discord.
It's the latest message right now, and can be found on the mister fpga discord, midi room(?)

**Comment by zaphod77** [comment]
okay i have the program.  not sure how to compile it, though. does running this on the actual rom produce a mkII with capital tone fallback that works with this emulator?

and what about the force gm reset as gs bit?  i know the correct thing to do is fix the broken midi files, but it's annoying that they playback just fine on a sound canvas soundfont, but not on a real sc-55

**Comment by Kitrinx** [comment]
the program I wrote was simply a tool to modify the rom so I could reprogram it and observe what adjusting various properties of the various instrument tables were, it can be modified in order to populate these tables in such a way that it will fall back, by making all invalid (blank) table entries the same as the first entry

**Comment by zaphod77** [comment]
actually, it should fallback to the highest variation less than or equal to the requested.  so it's not quite that simple.  the manuals that mention fallback clearly describe that, and that's what happens on pretty much every non roland card, as well as all the sf2 soundsynths.

and i'm not allowed to talk on that server, because it requires phone verification, which i had to turn off just to join a different server.  My phone number is somehow banned from severs i've never been on, and removing that verification let me onto them.

Bottom line is i want sc-55 mk1 logic with the mkII instrument map.

**Comment by Kitrinx** [comment]
If you join it and message me I will verify you

**Comment by zaphod77** [comment]
had to send a friend request. direct messages are blocked.

**Comment by Kappa971** [comment]
It should work with the "Capital Tone Fallback" option in Falcosoft MIDI Player with "Roland SCVA-55" map.
In theory, you should have two LoopMIDI ports, a "LoopMIDI In" port that all games will rely on, and a "LoopMIDI Out" port that the Nuked-SC55 will be connected to. Falcosoft MIDI Player should receive the midi stream from "LoopMIDI In" and send it to "LoopMIDI Out" (Nuked SC-55).
I haven't tried it yet but I will soon because I read that many games use CTF. It should work.

It would be nice if a built-in emulator option was added in the future, selectable with SC-55 models without CTF support as a workaround.
